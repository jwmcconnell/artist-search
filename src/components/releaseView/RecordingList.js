import React from 'react';
import PropTypes from 'prop-types';

import RecordingItem from './RecordingItem';

const styles = {
  list: {
    listStyle: 'none'
  }
};

const RecordingList = ({ recordingsData, artist }) => {
  const recordings = recordingsData.map(recording => (
    <RecordingItem key={recording.id} recording={recording} artist={artist} />
  ));
  return (
    <ul style={styles.list}>{recordings}</ul>
  );
};

RecordingList.propTypes = {
  recordingsData: PropTypes.array.isRequired,
  artist: PropTypes.string.isRequired
};

export default RecordingList;
