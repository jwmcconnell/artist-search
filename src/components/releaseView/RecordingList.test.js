import React from 'react';
import { shallow } from 'enzyme';
import RecordingList from './RecordingList';

describe('RecordingList component', () => {
  it('renders RecordingList', () => {
    const wrapper = shallow(<RecordingList recordingsData={[]} artist="jinsang" />);
    expect(wrapper).toMatchSnapshot();
  });
});
