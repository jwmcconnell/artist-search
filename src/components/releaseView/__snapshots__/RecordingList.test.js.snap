// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingList component renders RecordingList 1`] = `
<ul
  style={
    Object {
      "listStyle": "none",
    }
  }
/>
`;
